src/oom/oomd.c

```diff
@@ -0,0 +1,113 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <getopt.h>
+
+#include "daemon-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "oomd-manager.h"
+#include "parse-util.h"
+#include "pretty-print.c"
+#include "signal-util.h"
+
+static enum {
+        ACTION_RUN,
+        ACTION_HELP,
+        ACTION_DRY_RUN,
+} arg_action = ACTION_RUN;
+
+static int help(void) {
+        _cleanup_free_ char *link = NULL;
+        int r;
+
+        r = terminal_urlify_man("systemd-oomd", "1", &link);
+        if (r < 0)
+                return log_oom();
+
+        printf("%s [OPTIONS...]\n\n"
+                        "Process system password requests.\n\n"
+                        "  -h --help     Show this help\n"
+                        "     --dry-run  Log actions to be taken instead of killing\n"
+                        "\nSee the %s for details.\n"
+                        , program_invocation_short_name
+                        , link
+              );
+
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+        enum {
+                ARG_DRY_RUN,
+        };
+
+        static const struct option options[] = {
+                { "help",              no_argument,       NULL, 'h'                   },
+                { "dry-run",           no_argument,       NULL, ARG_DRY_RUN           },
+                {}
+        };
+
+        int c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "h", options, NULL)) >= 0)
+
+                switch (c) {
+
+                        case 'h':
+                                arg_action = ACTION_HELP;
+                                break;
+
+                        case ARG_DRY_RUN:
+                                arg_action = ACTION_DRY_RUN;
```

Comment:
This seems rather strange to set "dry run" as an action... The flag should apply to any action (and I'm assuming that there will be more actions in the future).